components/dashboard/src/Login.tsx

```diff
@@ -46,17 +48,46 @@ export function hasVisitedMarketingWebsiteBefore() {
 export function Login() {
     const { setUser } = useContext(UserContext);
     const { setTeams } = useContext(TeamsContext);
-    const showWelcome = !hasLoggedInBefore() && !hasVisitedMarketingWebsiteBefore();
+    const experiment = "login-from-context-6826";
 
-    const [ authProviders, setAuthProviders ] = useState<AuthProviderInfo[]>([]);
-    const [ errorMessage, setErrorMessage ] = useState<string | undefined>(undefined);
+    const urlHash = getURLHash();
+    let hostFromContext: string | undefined;
+    let repoPathname: string | undefined;
+
+    try {
+        if (urlHash.length > 0) {
+            const url = new URL(urlHash);
+            hostFromContext = url.host;
+            repoPathname = url.pathname;
+        }
+    } catch (error) {
+        // Hash is not a valid URL
+    }
+
+    const [authProviders, setAuthProviders] = useState<AuthProviderInfo[]>([]);
+    const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+    const [providerFromContext, setProviderFromContext] = useState<AuthProviderInfo>();
+
+    const showWelcome = Experiment.has(experiment) ?
+        (!hasLoggedInBefore() && !hasVisitedMarketingWebsiteBefore() && !urlHash.startsWith("https://"))
+        : (!hasLoggedInBefore() && !hasVisitedMarketingWebsiteBefore())
+        ;
 
     useEffect(() => {
         (async () => {
             setAuthProviders(await getGitpodService().server.getAuthProviders());
         })();
     }, [])
 
+    if (Experiment.has(experiment)) {
```

Comment:
:+1:

Reply:
@laushinka  My bad, turns our React does not like this: build failed with `Line 83:9:  React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks`.

Then the `if(Experiment.hat...` goes into the `useEffect`, on top-level. :+1:

Reply:
Yes just saw this as well